yaml.go

```diff
@@ -319,24 +318,10 @@ func failf(format string, args ...any) {
 	panic(&yamlError{fmt.Errorf("yaml: "+format, args...)})
 }
 
-// ParserError represents a fatal error encountered during the parsing phase.
-// These errors typically indicate a syntax issue in the YAML document that
-// prevents further processing.
-type ParserError struct {
-	Message string
-	Line    int
-	Column  int
-}
-
-func (e *ParserError) Error() string {
-	var b strings.Builder
-	b.WriteString("yaml: ")
-	if e.Line != 0 {
-		b.WriteString("line " + strconv.Itoa(e.Line) + ": ")
-	}
-	b.WriteString(e.Message)
-	return b.String()
-}
+type ParserError = libyaml.ParserError
+type ScannerError = libyaml.ScannerError
+type ReaderError = libyaml.ReaderError
+type WriterError = libyaml.WriterError
```

Comment:
I'm a bit annoyed because you are exporting all out internal type of error.

It could limit us in refactoring later.

I don't think user need to have access to every field of our error.

I would have defined sentinel errors, and return them by adding an ErrorIs on each type.

It's better for us and for users. They can simply catch 4 sentinel errors, and maybe we could wrap another common error witg them.

Reply:
This was the easiest way I found to keep the `TestParserErrorDecoder` test working. Feel free to make any changes to my PR (I don't think I fully understand what you are suggesting).